include/nn/socket.h

```diff
@@ -6,21 +6,35 @@
 #pragma once
 
 #include <nn/types.h>
+#include <nn/util.h>
 #include <sys/socket.h>
 
-namespace nn {
-namespace socket {
+#if NN_SDK_VER >= NN_MAKE_VER(7, 0, 0)
+namespace nn::socket {
 struct InAddr {
     u32 addr;
 };
+}  // namespace nn::socket
+#else
+struct in_addr;
```

Comment:
This could have ``using InAddr = ::in_addr`` and the other ifdef could be discarded, but I think either is fine for the repo. If the change isn't made in a day or 2 I'll just merge.

Reply:
Not sure if that would change the compiler's behavior. Let me test it out

Reply:
linker appears to be exporting the correct symbols with the alias